ublox/src/ubx_packets/packets.rs

```diff
@@ -379,7 +383,19 @@ impl fmt::Debug for FixStatusInfo {
     }
 }
 
-#[derive(Copy, Clone, Debug)]
+impl serde::Serialize for FixStatusInfo {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_map(Some(2))?;
+        state.serialize_entry("has_pr_prr_correction", &self.has_pr_prr_correction())?;
+        state.serialize_entry("map_matching", &self.map_matching())?;
+        state.end()
+    }
+}
```

Comment:
I think all newtypes like `pub struct FixStatusInfo(u8);` and bitflags should be serialized as plain ints. Things like `has_pr_prr_correction` are not actual fields on the messages, its just information derived from the fields.

For a newtype should be able to do `self.0.serialize(...)` or you could

```rust
/// Fix Status Information
#[repr(transparent)]
#[derive(Copy, Clone, serde::Serialize)]
#[serde(transparent)]
pub struct FixStatusInfo(u8);
```

For the flags it looks like the `bitflags!` macro generates a `.bits()` method so

```rust
gps_file.bits().serialize(...)
```

Reply:
This is how our current `x2json` tools work - they just have a field called `flags` or whatever